### PR TITLE
Implement all

### PR DESCRIPTION
**Pull request by Resci**

### CODE REVIEW

File: src/main/java/com/mate/dto/CinemaHallResponseDto.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.mate.dto;
+
+public class CinemaHallResponseDto {
+    private Long id;
+    private String description;
+
```
Comment (RuslanIbIsh) [inline src/main/java/com/mate/dto/CinemaHallResponseDto.java:6]: capacity?

File: src/main/java/com/mate/service/impl/CinemaHallServiceImpl.java [hunk 2]
```diff
@@ -19,6 +19,11 @@ public CinemaHall add(CinemaHall cinemaHall) {
         return cinemaHallDao.add(cinemaHall);
     }
 
+    @Override
+    public CinemaHall get(Long id) {
+        return cinemaHallDao.get(id).get();
```
Comment (RuslanIbIsh) [inline src/main/java/com/mate/service/impl/CinemaHallServiceImpl.java]: should not use get() from Optional without checks

File: src/main/java/com/mate/util/mapper/CinemaHallMapper.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        cinemaHall.setDescription(dto.getDescription());
+        cinemaHall.setCapacity(dto.getCapacity());
+        return cinemaHall;
+    }
+
+    @Override
+    public CinemaHallResponseDto mapToDto(CinemaHall cinemaHall) {
+        CinemaHallResponseDto responseDto = new CinemaHallResponseDto();
+        responseDto.setId(cinemaHall.getId());
+        responseDto.setDescription(cinemaHall.getDescription());
```
Comment (RuslanIbIsh) [inline src/main/java/com/mate/util/mapper/CinemaHallMapper.java]: capacity)))

File: src/main/java/com/mate/config/AppConfig.java [hunk 4]
```diff
@@ -14,7 +14,8 @@
 @PropertySource("classpath:db.properties")
 @ComponentScan(basePackages = {
         "com.mate.dao",
-        "com.mate.service"
+        "com.mate.service",
+        "com.mate.util.mapper"
```
Comment (RuslanIbIsh) [inline src/main/java/com/mate/config/AppConfig.java]: I don't know exectly - but in checklists was about location of package "mapper"

File: src/main/java/com/mate/controller/MovieSessionController.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionMapper movieSessionMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionMapper = movieSessionMapper;
+    }
+
+    @PostMapping
+    public MovieSessionResponseDto add(@RequestBody MovieSessionRequestDto requestDto) {
+        MovieSession movieSession;
```
Comment (RuslanIbIsh) [inline src/main/java/com/mate/controller/MovieSessionController.java]:
```suggestion
        MovieSession movieSession = null; // cosa locale variable  
```

File: src/main/java/com/mate/controller/MovieSessionController.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    public MovieSessionResponseDto update(@PathVariable Long id,
+                                          @RequestBody MovieSessionRequestDto requestDto) {
+        MovieSession movieSession = movieSessionMapper.mapToObj(requestDto);
+        movieSession.setId(id);
+        movieSessionService.update(movieSession);
+        return movieSessionMapper.mapToDto(movieSession);
+    }
+
+    @DeleteMapping("/{id}")
+    public String delete(@PathVariable Long id) {
```
Comment (RuslanIbIsh) [inline src/main/java/com/mate/controller/MovieSessionController.java]:
```suggestion
    public void delete(@PathVariable Long id) {
```

**Review by RuslanIbIsh — CHANGES_REQUESTED** [review]

**Review by RuslanIbIsh — APPROVED** [review]

File: src/main/java/com/mate/dto/CinemaHallResponseDto.java (hunk 1 above)
Comment (boroda4436) [reply src/main/java/com/mate/dto/CinemaHallResponseDto.java:6]: agree, we can add capacity to response dto

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**